### PR TITLE
Fix/stored xss in asset creation

### PR DESCRIPTION
**Pull request by adriangohjw**
## Problem

Stored XSS via arbitrary file upload Content-Type: the upload flow used client-controlled `Content-Type` and `Content-Disposition` when uploading to the presigned URL. An attacker could upload a file with an allowed extension (e.g. `.jpg`) but intercept the request and set `Content-Type: text/html` or `image/svg+xml`, causing the browser to execute script when the asset is viewed.

## Solution

Derive `Content-Type` and `Content-Disposition` from the validated filename extension (allowlist) in the asset service; pass them into the S3 presigned PUT URL so they are signed.

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [x] No - this PR is backwards compatible

## Tests

refer to the slack channel

---

> [!NOTE]
> **Medium Risk**
> Changes the asset upload contract by server-signing and enforcing `Content-Type`/`Content-Disposition` on presigned S3 PUTs; mis-matches could break uploads for some file types. Touches security-sensitive upload metadata handling, but scope is limited to the asset presign+upload path.
> 
> **Overview**
> **Hardens asset uploads against stored XSS/type confusion** by moving `Content-Type` and `Content-Disposition` derivation to the server and requiring the client to use those exact values when PUT-ing to S3.
> 
> The presigned PUT URL generation now includes and signs `ContentType`/`ContentDisposition` (via `signableHeaders`), the `asset.getPresignedPutUrl` response returns these fields, and the upload hook uses them instead of client-controlled metadata. Tests and MSW mocks are updated, and new unit coverage is added for MIME/content-disposition derivation (including Unicode edge cases).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 5a034529c5f6e3d5cc09bd8aeb953a16e63f2104. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: apps/studio/src/server/modules/asset/asset.service.ts [hunk 1]
```diff
@@ -6,12 +6,43 @@ import filenamify from "filenamify"
...
+}
+
+/**
+ * Derive trusted Content-Type from key. Key is only produced after schema validation,
+ * so the file extension is always from the allowlist.
+ */
+export const getContentTypeFromKey = (key: string): string => {
+  const segment = key.split("/").pop() ?? ""
+  const ext = segment.includes(".")
+    ? segment.toLowerCase().substring(segment.lastIndexOf("."))
```
[the remark this answers is not in the document]
Comment (adriangohjw) [reply apps/studio/src/server/modules/asset/asset.service.ts]: fix in https://github.com/opengovsg/isomer/pull/1839/commits/7f0677302f8548685739aed5f9cdd8fbcc27eb21

File: apps/studio/src/server/modules/asset/asset.service.ts [hunk 2]
```diff
@@ -6,12 +6,42 @@ import filenamify from "filenamify"
...
+
+/**
+ * Derive trusted Content-Type from key. Key is only produced after schema validation,
+ * so the file extension is always from the allowlist.
+ */
+export const getContentTypeFromKey = (key: string): string => {
+  const segment = key.split("/").pop() ?? ""
+  const lower = segment.toLowerCase()
+  const ext = lower.includes(".") ? lower.substring(lower.lastIndexOf(".")) : ""
+  return EXTENSION_TO_MIME[ext] ?? "application/octet-stream"
```
[the remark this answers is not in the document]
Comment (adriangohjw) [reply apps/studio/src/server/modules/asset/asset.service.ts]: the filename is validated upstream in the schema against the list of allowed extensions, so the extension at this step should be valid
